### ISSUE TITLE
when i function deploy in visual code `Service Unavailable`

### ISSUE THREAD

**Issue by piyushkacha7** [Microsoft/vscode-azurefunctions#1199]
Repro steps:1.2.Action: azureFunctions.deployError type: 503Error Message: Service UnavailableVersion: 0.16.0OS: win32VS Code Version: 1.33.1<details><summary>Call Stack</summary>```module.exports.s.pipeline (C:/Users/Kevit dist/extension.bundle.js:333:833828)s (C:/Users/Kevit dist/extension.bundle.js:277:110820)s (C:/Users/Kevit dist/extension.bundle.js:165:25754)C:/Users/Kevit dist/extension.bundle.js:165:26994o (C:/Users/Kevit dist/extension.bundle.js:223:7055)C:/Users/Kevit dist/extension.bundle.js:165:24675$.module.exports.t.requestLibrarySink.t.headersOnly.t.streamedResponse.t.body.function.t.body.pipe.a [as _callback] (C:/Users/Kevit dist/extension.bundle.js:1:579051)$.module.exports.$.init.t._callback.t.callback.t.callback (C:/Users/Kevit dist/extension.bundle.js:253:5461)$.emit (events.js:182:13)$.<anonymous> (C:/Users/Kevit dist/extension.bundle.js:253:18274)$.emit (events.js:187:15)IncomingMessage.<anonymous> (C:/Users/Kevit dist/extension.bundle.js:253:17116)Object.onceWrapper (events.js:273:13)IncomingMessage.emit (events.js:187:15)endReadableNT (_stream_readable.js:1090:12)process._tickCallback (internal/process/next_tick.js:63:19)```</details>

**Comment by EricJizbaMSFT** [comment]
@piyushkacha7 can you add some detailed repro steps to this issue? Also - "Service Unavailable" will sometimes happen and there's not much we can do about it in the VS Code extension. If it's persistent definitely let us know, but otherwise try deploying again and hopefully that will resolve the issue.

**Comment by piyushkacha7** [comment]
@EricJizbaMSFT I created my resource group from the Azure portal also created an azure function application form portal because of VS code don't support to change the default hosting plan. 
so first I create an app in the portal and try to deploy from VS code then give me this kind of error
and currently I don't see my Streaming log in VC code.

**Comment by EricJizbaMSFT** [comment]
What type of function app did you create in the portal?
1. What runtime?
1. What region?
1. What kind of app service plan (Consumption vs. Dedicated)?

Do you have full permissions to that function app?

**Comment by piyushkacha7** [comment]
I'm using Following: 
Runtime: Linux  
Region: Central US
App service plan: Dedicated 

Also, I'm not able to connect to log streaming service and see the logs that I've done using context.log() function. Also, not able to connect to the log streaming service using VS Code.

**Comment by EricJizbaMSFT** [comment]
Okay I can't repro "Service Unavailable", but I _can_ repro the fact that streaming logs doesn't work for a linux dedicated plan. Updating title to reflect that.

As a workaround, you should be able to stream logs here:
https://\<insert site name\>.scm.azurewebsites.net/api/logstream

In general, this site just lets you view a lot more logs and advanced tools:
https://\<insert site name\>.scm.azurewebsites.net

**Comment by piyushkacha7** [comment]
Hello Mr.Eric Jizba, I want to know how much time for this issue slove? because we need to see all the log of our azure function as soon as possible.

**Comment by EricJizbaMSFT** [comment]
@piyushkacha7 did you try going to the site I mentioned? You should be able to see log stream either in that site or in the portal. I don't have an estimate for when we fix this is VS Code, but I'm hoping you're not blocked since you should be able to see the logs there.

**Comment by piyushkacha7** [comment]
@EricJizbaMSFT 
yes, I can see log **https://<insert site name>.scm.azurewebsites.net/api/logstream** but in this only log like `[40m[32minfo[39m[22m[49m: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
2019-06-12T05:21:13.800889850Z: [INFO]        Request finished in 357.8865ms 200 application/json; charset=utf-8` request success or request failed. 
we are a developer so need see log if any request failed then failed log error need to see.
any proper way for I can see my full logs? in the above site only display logs like success or failure. but I need to see debug logs. before stopping the VS Code stream its display fully logs so it's easy to debug application error. 

` context.log('HTTP trigger function processed a request.');
 context.log.error(`Exception`);
 context.log.error(error);`

I need to see these logs live tracking. it's easy to debug me.
So any other way to see my fully debug and error logs?

**Comment by EricJizbaMSFT** [comment]
@piyushkacha7 the best I can do from VS Code is replicate the experience you see in that site. That's my plan to fix this issue.

If you want to see your console logs you would have to file an issue directly with the Azure Functions team. See here for more info: https://github.com/Azure/azure-functions#questions--help
